client/src/use-query.tsx

```diff
@@ -0,0 +1,26 @@
+import {Entity, generateZQL} from '@rocicorp/rails';
+import {EntityQuery} from '@rocicorp/rails/out/zql/query/entity-query.js';
+import {EntitySchema} from '@rocicorp/rails/out/zql/schema/entity-schema.js';
+import {useEffect, useMemo, useState} from 'react';
+
+export function useQuery<S extends EntitySchema, R>(
+  query: EntityQuery<S, R>,
+  dependencies: unknown[] = [],
+): R {
+  const view = useMemo(() => query.prepare().materialize(), dependencies);
+  const [value, setValue] = useState<R>(view.value as R);
+  useEffect(() => {
+    setValue(view.value);
+    return view.on(setValue);
+  }, dependencies);
+  return value;
+}
+
+type ReplicacheLike = Parameters<typeof generateZQL>[0];
+
+export function useTable<E extends Entity>(
```

Comment:
A few things:

* Is this a hook? `use` is typically reserved for hooks.
* I don't think we'll call them tables, probably collections.
* What about `query<T>(rep, name)`
* Also I think it is fine to wrap this all up into a `useQuery` react hook and not expose `useTable`/`query` at all. I bet it will feel more natural to just repeat shared sections of the query rather than share them. They're so succint already.

Reply:
`useTable` does feel redundant.

Reply:
I initially had a useMemo in there... that is why it was structured like a hook.

Reply:
> Also I think it is fine to wrap this all up into a useQuery react hook and not expose useTable/query at all. I bet it will feel more natural to just repeat shared sections of the query rather than share them. They're so succint already.

The query starts with an object... If we want to collapse this into one hook it needs to be something along the lines of:

```ts
useQuery(rep, 'list', q => q.select('id', 'name').where(...));
```

This looks fine to me :-)